components/p3a/brave_p3a_config.cc

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/p3a/brave_p3a_config.h"
+
+#include "base/command_line.h"
+#include "base/logging.h"
+#include "brave/components/p3a/brave_p3a_switches.h"
+#include "brave/components/p3a/buildflags.h"
+
+namespace brave {
+
+namespace {
+
+constexpr uint64_t kDefaultUploadIntervalSeconds = 60;  // 1 minute.
+
+void LoadTimeDelta(base::CommandLine* cmdline,
+                   const char* switch_name,
+                   base::TimeDelta* result) {
+  if (cmdline->HasSwitch(switch_name)) {
+    std::string seconds_str = cmdline->GetSwitchValueASCII(switch_name);
+    int64_t seconds;
+    if (base::StringToInt64(seconds_str, &seconds) && seconds > 0) {
+      *result = base::Seconds(seconds);
+    }
+  }
+}
+
+void LoadString(base::CommandLine* cmdline,
+                const char* switch_name,
+                std::string* result) {
+  if (cmdline->HasSwitch(switch_name)) {
+    *result = cmdline->GetSwitchValueASCII(switch_name);
+  }
+}
+
+void LoadURL(base::CommandLine* cmdline,
```

Comment:
this name is very confusing because it sounds like you are making a request to the url. Also do not use out params for this